docs/hash-set.rst

```diff
@@ -0,0 +1,386 @@
+``s``, replacing an existing equal element if it already exists.
+
+::
+
+    HashSet.insert 100 HashSet.empty
+    > fromList [100]
+
+    HashSet.insert 0 (HashSet.fromList [0, 2, 4, 6])
+    > fromList [0,2,4,6]
+
+Removing an element from a set
+""""""""""""""""""""""""""""""
+
+::
+
+    HashSet.delete :: a -> HashSet a -> HashSet a
+    HashSet.delete x s = ...
+
+:haddock_short:`/Data.HashSet#delete` the element ``x`` from the set ``s``. If
+it’s not a member it leaves the set unchanged.
+
+::
+
+    HashSet.delete 0 (HashSet.fromList [0, 2, 4, 6])
+    > fromList [2,4,6]
+
+Filtering elements from a set
+"""""""""""""""""""""""""""""
+
+::
+
+    HashSet.filter :: (a -> Bool) -> HashSet a -> HashSet a
+    HashSet.filter predicate s = ...
+
+:haddock_short:`/Data.HashSet#filter` produces a set consisting of all elements
+of ``s`` for which the ``predicate`` returns ``True``.
+
+::
+
+    HashSet.filter (==0) (HashSet.fromList [0, 2, 4, 6])
+    > fromList [0]
+
+
+Set Operations
+^^^^^^^^^^^^^^
+
+Union
+"""""
+
+::
+
+    HashSet.union :: HashSet a -> HashSet a -> HashSet a
+    HashSet.union l r = ...
+
+:haddock_short:`/Data.HashSet#union` returns a set containing all elements that
+are in either of the two sets ``l`` or ``r`` (`set union
+<https://en.wikipedia.org/wiki/Union_(set_theory)>`_).
+
+::
+
+    HashSet.union HashSet.empty (HashSet.fromList [0, 2, 4, 6])
+    > fromList [0,2,4,6]
+
+    HashSet.union (HashSet.fromList [1, 3, 5, 7]) (HashSet.fromList [0, 2, 4, 6])
+    > fromList [0,1,2,3,4,5,6,7]
+
+Intersection
+""""""""""""
+
+::
+
+    HashSet.intersection :: HashSet a -> HashSet a -> HashSet a
+    HashSet.intersection l r = ...
+
+:haddock_short:`/Data.HashSet#intersection` returns a set the elements that are
+in both sets ``l`` and ``r`` (`set intersection
+<https://en.wikipedia.org/wiki/Intersection_(set_theory)>`_).
+
+::
+
+    HashSet.intersection HashSet.empty (HashSet.fromList [0, 2, 4, 6])
+    > fromList []
+
+    HashSet.intersection (HashSet.fromList [1, 3, 5, 7]) (HashSet.fromList [0, 2, 4, 6])
+    > fromList []
+
+    HashSet.intersection (HashSet.singleton 0) (HashSet.fromList [0, 2, 4, 6])
+    > fromList [0]
+
+Difference
+""""""""""
+
+::
+
+    HashSet.difference :: HashSet a -> HashSet a -> HashSet a
+    HashSet.difference l r = ...
+
+:haddock_short:`/Data.HashSet#difference` returns a set containing the elements
+that are in the first set ``l`` but not the second set ``r`` (`set
+difference/relative compliment
+<https://en.wikipedia.org/wiki/Complement_(set_theory)#Relative_complement>`_).
+
+::
+
+    HashSet.difference (HashSet.fromList [0, 2, 4, 6]) HashSet.empty
+    > fromList [0,2,4,6]
+
+    HashSet.difference (HashSet.fromList [0, 2, 4, 6]) (HashSet.fromList [1, 3, 5, 7])
+    > fromList [0,2,4,6]
+
+    HashSet.difference (HashSet.fromList [0, 2, 4, 6]) (HashSet.singleton 0)
+    > fromList [2,4,6]
+
+
+Serialization
+-------------
+
+TODO(m-renaud): Write serialization section.
```

Comment:
Same here

Reply:
Removed TODO's, can add serialization section in a follow-up PR if desired.